### ISSUE TITLE
Crash on chunk split [2.1.1]

### ISSUE THREAD

**Issue by ttohin** [jonas/tig#418]
I'm in stage view. My diff contains a lot of small changes and I want to split them to several chunks.
Cursor is on the position like on screenshot.
![crash](https://cloud.githubusercontent.com/assets/4542136/8128507/0b7bd808-1100-11e5-8078-b4087d18cd36.png)

I pressed ```\``` and got crash with the message: 
```
tig(48324,0x7fff7cd2f300) malloc: *** error for object 0x7fc4e2806e08: incorrect checksum for freed object - object was probably modified after being freed.
*** set a breakpoint in malloc_error_break to debug
```
Environment: macosx, iterm2, tmux

**Comment by jonas** [comment]
Would it be possible to compile Tig with debug symbols using `make all-debug` and then run it inside a debugger in order to get a stacktrace?

**Comment by ttohin** [comment]
screenshot from tmux:
![crash_tig](https://cloud.githubusercontent.com/assets/4542136/8189466/c8f34754-145c-11e5-9f9f-ef9250ec6990.png)
and from xcode with stacktrace:
![crash_lldb](https://cloud.githubusercontent.com/assets/4542136/8189467/c8f36b76-145c-11e5-81e2-c3b30de7d499.png)

**Comment by ttohin** [comment]
Any ideas?

**Comment by jonas** [comment]
Since June I've added a test for split-chunk (see commit 012aea375016f87aa33f79396479a7101fd5ac89) which exposed several memory allocation issues that was fixed in 65cb5b4242bffe5950402289119d75aa1ef1678a. The test is quite simple (see [test/stage/split-chunk-test](https://github.com/jonas/tig/blob/3448d1cd6a3a3bedd0668c6b6b60fd545a3614ed/test/stage/split-chunk-test)) but it might also have fixed this issue.

When you have time please test current master.

**Comment by jonas** [comment]
@ttohin Closing this as I believe it has been fixed. See above comment.
